nixos/lib/make-disk-image.nix

```diff
@@ -284,6 +297,21 @@ in pkgs.vmTools.runInLinuxVM (
       # The above scripts will generate a random machine-id and we don't want to bake a single ID into all our images
       rm -f $mountPoint/etc/machine-id
 
+      # Set the ownerships of the contents. The modes are set in preVM.
+      # No globbing on targets, so no need to set -f
+      targets_=(${concatStringsSep " " targets})
+      users_=(${concatStringsSep " " users})
+      groups_=(${concatStringsSep " " groups})
+      for ((i = 0; i < ''${#targets_[@]}; i++)); do
+        target="''${targets_[$i]}"
+        user="''${users_[$i]}"
+        group="''${groups_[$i]}"
+        if [ -n "$user$group" ]; then
+          # We have to nixos-enter since we need to use the user and group of the VM
+          nixos-enter --root $mountPoint -- chown -R "$user:$group" "$target"
```

Comment:
Looks to me like this will behave unexpectedly if in particular only `user` is specified. Maybe add an assert to prevent cases where only one of user/group is being set?

Reply:
`chown` from `coreutils` supports both `user:` and `:group`. Do we have to support other `chown`s that don't allow that syntax?

Reply:
`:group` appears to work correctly, but when I tried `chown root: test-file` it also changed the group of `test-file` to `root`.

TBH, I'm not sure we have to support other `chown`s that don't allow that syntax, but it feels to me like restricting our API surface to say “you're allowed to chown iff you set both user and group” wouldn't be a big problem in practice (as most of the time one would want to do both), and it might help avoiding unintuitive behavior?

Reply:
You're right, I missed this part of the chown docs:

> If a colon but no group name follows the user name, that user is  made  the
> owner  of  the files **and the group of the files is changed to that user's login group**

At this point couldn't we just add another if and mirror the etc.*.uid/gid behaviour? I don't think this will have any problematic cases:

```sh
if [ -n $group ]; then
  chown $user:$group $file
else
  if [ -n $user ]; then
    chown $user $file
  fi
fi
```

Reply:
That's totally possible too, if you think there's an actual use case for people setting only one of user/group — I'm not sure it'd bring much value and it expands the API we then have to maintain, but why not.

Reply:
I have no real use case in mind — I'll leave the API as is and document the limited options then.